frontend/src/static/js/main.js

```diff
@@ -3,16 +3,32 @@ window.REPOS = {};
 
 window.hashParams = {};
 window.addHash = function addHash(newKey, newVal) {
+  window.hashParams[newKey] = newVal;
+};
+window.rmHash = function rmHash(key) {
+  window.hashParams[key] = null;
+};
+
+window.encodeHash = function encodeHash() {
   const { hashParams } = window;
-  hashParams[newKey] = newVal;
 
-  const hash = [];
-  const enquery = (key, val) => `${key}=${encodeURIComponent(val)}`;
-  Object.keys(hashParams).forEach((hashKey) => {
-    hash.push(enquery(hashKey, hashParams[hashKey]));
-  });
+  window.location.hash = Object.keys(hashParams)
+      .map((key) => (hashParams[key]
+        ? `${key}=${encodeURIComponent(hashParams[key])}` : ''))
+      .filter((x) => x).join('&');
```

Comment:
Why not filter out removed params in `rmHash`? Since this also means that no key can contain a `false` boolean value.

Reply:
gd catch